firewood/src/merkle/node.rs

```diff
@@ -3,27 +3,41 @@
 
 use crate::shale::{disk_address::DiskAddress, CachedStore, ShaleError, ShaleStore, Storable};
 use bincode::{Error, Options};
+use bitflags::bitflags;
 use enum_as_inner::EnumAsInner;
 use serde::{de::DeserializeOwned, Deserialize, Serialize};
 use sha3::{Digest, Keccak256};
 use std::{
-    fmt::{self, Debug},
+    fmt::Debug,
     io::{Cursor, Read, Write},
     sync::{
         atomic::{AtomicBool, Ordering},
         OnceLock,
     },
 };
 
+mod branch;
+mod extension;
+mod leaf;
+mod partial_path;
+
+pub use branch::{BranchNode, MAX_CHILDREN, SIZE as BRANCH_NODE_SIZE};
```

Comment:
pub(crate) or pub(super). We don't want these as part of the public API.

Reply:
https://github.com/ava-labs/firewood/issues/362

It's a bit of a tangled web. I will do this later if it's alright with you. The `node` mod itself isn't public, so I don't think any of these can be accessed outside the crate, but I agree that we should be more strict in case we ever change the visibility of the `node` mod.